src/modules/flow/iio/nodes.c

```diff
@@ -71,24 +145,82 @@ reader_cb(void *data, struct sol_iio_device *device)
 }
 
 static bool
-create_device_channels(struct gyroscope_data *mdata, int device_id)
+create_device_channels(struct iiodevice_data *mdata, int device_id, enum iiodevice_types type)
 {
     struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
-
+    int i = 0;
+ 
     mdata->device = sol_iio_open(device_id, &mdata->config);
     SOL_NULL_CHECK(mdata->device, false);
 
 #define ADD_CHANNEL(_axis) \
```

Comment:
indeed, an explicit `ADD_CHANNEL(in_anglvel_x, 0); ADD_CHANNEL(in_anglvel_y, 1); ...` would be shorter and cleaner.